src/WpfMath/UserDefinedColorParser.cs

```diff
@@ -0,0 +1,264 @@
+            /// <summary>
+            /// Type is a Hexadecimal <see cref="string"/> consisting of Hexadecimal values of the color.<para/>
+            /// Alpha channel is strictly = FF.
+            /// </summary>
+            Hex_shortString,
+
+
+            /// <summary>
+            /// An unrecognized ColorStringType.
+            /// </summary>
+            None
+        }
+
+        /// <summary>
+        /// Gets the type of <see cref="ColorStringTypes"/> from the <paramref name="input"/>.
+        /// </summary>
+        /// <param name="input"></param>
+        /// <returns></returns>
+        private static ColorStringTypes GetColorString(string input)
+        {
+            if (IsByteHexTrain(input,6)==true)
+            {
+                return ColorStringTypes.Hex_shortString;
+            }
+            else if (IsByteHexTrain(input,8))
+            {
+                return ColorStringTypes.Hex_longString;
+            }
+            else if (IsByteTrain(input, 3) == true)
+            {
+                return ColorStringTypes.Byte_shortString;
+            }
+            else if (IsByteTrain(input, 4) == true)
+            {
+                return ColorStringTypes.Byte_longString;
+            }
+            else
+            {
+                return ColorStringTypes.None;
+            }
+        }
+
+        /// <summary>
+        /// Returns a value that tells if the <paramref name="input"/> contains byte values, separated by a ",".
+        /// </summary>
+        /// <example>
+        /// 23,78,56=>true
+        /// 789,fp=>false
+        /// </example>
+        /// <param name="input"></param>
+        /// <param name="num"></param>
+        /// <returns></returns>
+        private static bool IsByteTrain(string input, int num)
+        {
+            bool StrCheck = false;
+            string[] arrStr = input.Trim().Split(',');
+            if (num == arrStr.Length)
+            {
+                int i = 0;
+                foreach (var item in arrStr)
+                {
+                    if (Byte.TryParse(item, out byte result) == true)
+                    {
+                        i += 1;
+                    }
+                    else { continue; }
+                }
+                StrCheck = i == num ? true : false;
+            }
+            else
+            {
+                StrCheck = false;
+            }
+            return StrCheck;
+        }
+        
+        /// <summary>
+        /// Returns a value that tells if the <paramref name="input"/> contains byte hex values,
+        /// </summary>
+        /// <param name="input"></param>
+        /// <param name="num"></param>
+        /// <returns></returns>
+        /// <remarks><paramref name="input"/> should be left in the raw state(e.g.;#56e245, not 56e245).</remarks>
+        private static bool IsByteHexTrain(string input, int num)
+        {
+            bool StrCheck = false;
+            string subStr = input.Substring(1);
```

Comment:
Empty string will fail here!

Reply:
The catch block handles that

Reply:
Well, "catch-all" clause could catch chat. But it's a widely recognized antipattern, we definitely don't want to catch everything in our code!